### ISSUE TITLE
Error with using the Gather Essence recipe

### ISSUE THREAD

**Issue by DrTravisProthro** [AngryBeaver/beavers-crafting#137]
I cannot get the gather essence recipe to work as intended where it pulls the item's rarity and gives x amount of essence based on the rarity.

It errors with a type error and says it cannot convert undefined or null to an object. Not exactly sure what I am doing wrong or if there is simply a conflict with another module or the dnd5e system itself.

![image](https://github.com/AngryBeaver/beavers-crafting/assets/52718054/22598b60-3f8c-4d8a-a3bc-c03482c5e4c8)
![image](https://github.com/AngryBeaver/beavers-crafting/assets/52718054/91f6233f-c09e-48e9-8cbc-fc8b7226cd1c)

**Comment by AngryBeaver** [comment]
Yeah the potions module is outdated. The internal structure for recipes had changed.
When you open up the recipe go to the advanced section and edit the macro the "for" loop your will encounter there needs to look like this instead: 
for(const ingredient of Object.values(Object.values(recipeData.input)[0])){

also in dnd5e it seems the anyOfPotion is also not working anymore the new subtype detection for potions should look like:
![image](https://github.com/AngryBeaver/beavers-crafting/assets/10677192/d8990bda-86c3-4725-a8c6-56a2b94e8d2b)

**Comment by DrTravisProthro** [comment]
Thank you for thr reply. I could see the error happening but coding isn't my background so didn't know how to fix it.

I did figure out a workaround for the AnyOfPotions however. I set the second condition to use item.system.source.unqiue === "Potion" and then simply tagged all my potions with the unique source and it worked like a charm.
